### PR TITLE
Use S3 client for uplooads/downloads during perf test

### PR DESCRIPTION
**Pull request by krishnasrinivas**
## Description
Use S3 client to upload/download object during perf tests. This reflects a more real world workload.

## How to test this PR?
mc support perf object ALIAS

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/perf-tests.go [hunk 1]
```diff
@@ -61,6 +62,14 @@ func selfSpeedtest(ctx context.Context, size, concurrent int, duration time.Dura
 	var totalBytesWritten uint64
 	var totalBytesRead uint64
 
+	client, err := minio.NewCore(globalLocalNodeName, &minio.Options{
+		Creds:  credentials.NewStaticV4(globalActiveCred.AccessKey, globalActiveCred.SecretKey, ""),
+		Secure: globalIsTLS,
+	})
+	if err != nil {
+		return SpeedtestResult{}, err
+	}
```
Comment (harshavardhana) [inline cmd/perf-tests.go]: you might also want to use the globalProxyTransport that trust all the certs etc. 

```
        opts := &minio.Options{
                Creds:     creds,
                Secure:    globalIsTLS,
                Transport: globalProxyTransport,
        }
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Looks like formatting is missing

```
cmd/perf-tests.go:66: File is not `gofmt`-ed with `-s` (gofmt)
		Creds:  credentials.NewStaticV4(globalActiveCred.AccessKey, globalActiveCred.SecretKey, ""),
		Secure: globalIsTLS,
                Transport: globalProxyTransport,
make: *** [Makefile:37: lint] Error 1
```

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by poornas — APPROVED** [review]

**Comment by harshavardhana** [comment]
Looks like this PR broke speedtest @krishnasrinivas 

First is a crash

```
[y4m4](https://app.slack.com/team/U8F589L58)  [11:21 PM](https://miniohq.slack.com/archives/D8FKNPZK6/p1647498098110059)
Mar 17 00:20:45 pfedlr750-1 minio[253118]: panic: "POST /minio/admin/v3/speedtest?autotune=true&concurrent=32&duration=10s&size=67108864": invalid WriteHeader code 0
Mar 17 00:20:45 pfedlr750-1 minio[253118]: goroutine 31042 [running]:
Mar 17 00:20:45 pfedlr750-1 minio[253118]: runtime/debug.Stack()
Mar 17 00:20:45 pfedlr750-1 minio[253118]: #011runtime/debug/stack.go:24 +0x65
Mar 17 00:20:45 pfedlr750-1 minio[253118]: github.com/minio/minio/cmd.setCriticalErrorHandler.func1.1()
Mar 17 00:20:45 pfedlr750-1 minio[253118]: #011github.com/minio/minio/cmd/generic-handlers.go:469 +0x8e
Mar 17 00:20:45 pfedlr750-1 minio[253118]: panic({0x217f740, 0xc02446a820})
Mar 17 00:20:45 pfedlr750-1 minio[253118]: #011runtime/panic.go:1038 +0x215
Mar 17 00:20:45 pfedlr750-1 minio[253118]: net/http.checkWriteHeaderCode(...)
Mar 17 00:20:45 pfedlr750-1 minio[253118]: #011net/http/server.go:1089
Mar 17 00:20:45 pfedlr750-1 minio[253118]: net/http.(*response).WriteHeader(0xc0240749a0, 0x0)
Mar 17 00:20:45 pfedlr750-1 minio[253118]: #011net/http/server.go:1123 +0x5c8
Mar 17 00:20:45 pfedlr750-1 minio[253118]: github.com/minio/minio/internal/logger.(*ResponseWriter).WriteHeader(0xc024070aa0, 0x0)
Mar 17 00:20:45 pfedlr750-1 minio[253118]: #011github.com/minio/minio/internal/logger/audit.go:116 +0x83
Mar 17 00:20:45 pfedlr750-1 minio[253118]: github.com/minio/minio/cmd.writeResponse({0x4a97630, 0xc024070aa0}, 0xc0013360c0, {0xc017a3d170, 0x0, 0x8f}, {0x26054b9, 0x10})
Mar 17 00:20:45 pfedlr750-1 minio[253118]: #011github.com/minio/minio/cmd/api-response.go:736 +0x225
Mar 17 00:20:45 pfedlr750-1 minio[253118]: github.com/minio/minio/cmd.writeErrorResponseJSON({0x4aba778, 0xc02404df80}, {0x4a97630, 0xc024070aa0}, {{0x0, 0x0}, {0x0, 0x0}, 0x0}, 0xc024047710)
Mar 17 00:20:45 pfedlr750-1 minio[253118]: #011github.com/minio/minio/cmd/api-response.go:816 +0x1aa
Mar 17 00:20:45 pfedlr750-1 minio[253118]: github.com/minio/minio/cmd.adminAPIHandlers.ObjectSpeedtestHandler({}, {0x4a97630, 0xc024070aa0}, 0xc024c00400)
Mar 17 00:20:45 pfedlr750-1 minio[253118]: #011github.com/minio/minio/cmd/admin-handlers.go:1060 +0xc1b
Mar 17 00:20:45 pfedlr750-1 minio[253118]: github.com/minio/minio/cmd.adminAPIHandlers.SpeedtestHandler(...)
Mar 17 00:20:45 pfedlr750-1 minio[253118]: #011github.com/minio/minio/cmd/admin-handlers.go:985
Mar 17 00:20:45 pfedlr750-1 minio[253118]: net/http.HandlerFunc.ServeHTTP(...)
Mar 17 00:20:45 pfedlr750-1 minio[253118]: #011net/http/server.go:2047
```

and another is we freeze incoming S3 API calls so this change would have never worked. Fixing this now.

**Pull request merged**